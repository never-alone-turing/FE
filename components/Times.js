import React from "react";
import { View, Text, StyleSheet, Button, TouchableOpacity } from "react-native";

const Times = ({item, deleteTimer}) => {
  console.log('item', item)
  return (
      <TouchableOpacity style={styles.listItem} >
        <View style={styles.listItemView}>
          <Button title='X' style={styles.closeButton} onPress={() => deleteTimer(item.id)}/>
          <Text style={styles.listItemName}>Name: {item.name}</Text>
          <Text style={styles.listItemName}>Time: {new Date(item.time).getHours().toString()}:{new Date(item.time).getMinutes().toString()}</Text>
          <Text style={styles.listItemName}>Window: {new Date(item.window).getHours().toString()}:{new Date(item.window).getMinutes().toString()}</Text>
        </View>
      </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  listItem: {
    padding: 15,
    backgroundColor: '#f8f8f8',
    borderBottomWidth: 1,   
    borderColor: "#eee"
  },
  listItemView: {
    flexDirection: 'column'
  },
  listItemName: {
    fontSize: 18
  },
  listItemTime: {
    fontSize: 18,
  },
  closeButton: {
    fontSize: 22,
    color: 'firebrick',
    backgroundColor: 'black'
  }
})

export default Times;