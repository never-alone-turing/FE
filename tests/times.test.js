import React from 'react'
import {create} from 'react-test-renderer'
import Times from '../components/Times'
import renderer from 'react-test-renderer'


  const mockItem = {
    name: 'User',
    time: '7:00',
    window: '15 minutes'
  }

  describe("<Times />", () => {

    it('renders correctly', () => {
      const tree = renderer.create(<Times
        item={"mockItem"}
        />).toJSON();
    expect(tree).toMatchSnapshot()
    })

  })
