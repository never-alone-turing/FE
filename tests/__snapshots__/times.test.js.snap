// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Times /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#f8f8f8",
      "borderBottomWidth": 1,
      "borderColor": "#eee",
      "opacity": 1,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "column",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          X
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Name: 
    </Text>
    <Text
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Time: 
    </Text>
    <Text
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Window: 
    </Text>
  </View>
</View>
`;
